src/awkward/_v2/index.py

```diff
@@ -1,79 +1,84 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+from __future__ import absolute_import
+
 import awkward as ak
 
+np = ak.nplike.NumpyMetadata.instance()
+
+
+class Index(object):
```

Comment:
If classes aren't inheriting from anything else, they need to inherit from `object` to avoid "old-style classes" in Python 2.